### PR TITLE
Allow clean removal of image mosaic files and support elements when removing from the granule store

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleStoreDecorator.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+ * GeoTools - The Open Source Java GIS Toolkit
+ * http://geotools.org
+ *
+ * (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleStoreDecorator.java]: -> 2018 :)

**Review by dromagnoli — APPROVED** [review]
Looks good to me.
I have only spotted a wrong copyright year for a new file.
Finally, it looks like the windows build failed :(

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleStoreDecorator.java (hunk 1 above)
Comment (aaime) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleStoreDecorator.java]: Ah yep

**Comment by aaime** [comment]
About the windows failure, the Slice2DIndexManager is indeed leaving RandomAccessFile instances open.. trying to figure out where and how.

**Comment by aaime** [comment]
Was due to un-closed readers and deferred loading. Also added machinery to indicate the issue happened and a flag to debug it (similar to the one available on the JDBCFeatureReader)

**Comment by aaime** [comment]
Phew, finally builds on Windows too... I'll squash all the extra commits and merge.

**Pull request closed without merging**